dissect/target/plugins/os/windows/cam.py

```diff
@@ -0,0 +1,354 @@
+            program_id (string): Program ID of application, unclear what this value means.
+            package_family_name (string): Package name of application using capability.
+            access_guid (string): Unclear what the value of this is.
+            label (string): Unclear what the value of this is, no joinable table with this ID.
+            app_name (string): Name of the application using capability.
+            binary_full_path (path): Path of the application using capability.
+            service_name (string): Name of the service using capability.
+
+        Record CamIdentityRelationshipHistoryRecord:
+
+        .. code-block:: text
+
+            last_observed_time (datetime): Last time capability was observed.
+            package_type (string): The application type of the record, originates from the table name.
+            file_id (string): The sha1 hash of the application making use of the capability.
+            file_id_hash (digest): Digest version of the file_id field.
+            program_id (string): Program ID of application, unclear what this value means.
+            binary_full_path (path): Path of the application using capability.
+
+        Record CamGlobalPromptHistoryRecord
+
+        .. code-block:: text
+
+            shown_time (datetime): Last time capability was observed.
+            package_type (string): The application type of the record, originates from the table name.
+            capability (string): The capability being used; microphone, camera, location etc.
+            file_id (string): The sha1 hash of the application making use of the capability.
+            file_id_hash (digest): Digest version of the file_id field.
+            program_id (string): Program ID of application, unclear what this value means.
+        """
+
+        self.camdb = self._open_db()
+
+        # Silently exit the plugin if no database object could be created.
+        # This means the plugin is not compatible with the target.
+        if not self.camdb:
+            return None
+
+        # Create mapping dict for sql 'join' actions.
+        self.MAPDB = self._build_context_dict()
+
+        # Iterate over the 'CONTENT_TABLES' and yield records
+        for table in self.CONTENT_TABLES:
+            for row in self.camdb.table(table):
+                # Resolve user if row contains a UserSid
+                if user := self.target.user_details.find(self.MAPDB["Users"].get(row["UserSid"], "")):
+                    user = user.user
+
+                # Create digest from FileID if field exists in row
+                if FileId_hash := self.MAPDB["FileIDs"].get(row["FileID"]):
+                    FileId_hash = digest((None, FileId_hash[4:], None))
+
+                # Yield record depending on table we are processing
+                if table in ["NonPackagedUsageHistory", "PackagedUsageHistory"]:
+                    last_used_time_stop = wintimestamp(row.get("LastUsedTimeStop", 0))
```

Comment:
As `wintimestamp(0)=datetime.datetime(1601, 1, 1, 0, 0, tzinfo=datetime.timezone.utc)`, I wonder if this then shows the year 1601 in the record for last_used_time_stop/start? I imagine that is confusing.

Perhaps it is better to deal with the `None` values instead.